yt_dlp/extractor/niconico.py

```diff
@@ -1055,7 +1072,10 @@ def _real_extract(self, url):
             }),
             'description': clean_html(traverse_obj(embedded_data, ('program', 'description'))),
             'timestamp': int_or_none(traverse_obj(embedded_data, ('program', 'openTime'))),
-            'is_live': True,
+            'live_status': live_status,
             'thumbnails': thumbnails,
-            'formats': formats,
+            'formats': [*self._yield_formats(ws, video_id, latency, live_status == 'is_live')] if ws else None,
+            'live_latency': latency,
+            'http_headers': headers,
+            '__ws': ws,
```

Comment:
Actually, the proper way is to put non-standard fields in `downloader_options`
```suggestion
            'http_headers': headers,
            'downloader_options': {
                'live_latency': latency,
                'ws': ws,
            },
```